### PR TITLE
feat: Support for pushing to two repositories

### PR DESCRIPTION
**Pull request by antroy-madetech**
Demodjango will need to push to both public and private repos. This PR enables that.

### CODE REVIEW

File: image_builder/commands/build.py [hunk 1]
```diff
@@ -39,6 +39,8 @@ def build(publish, send_notifications):
         )
 
         click.echo(f"Using ECR repository: {codebase.build.repository}")
+        # if codebase.build.additional_repository:
+        #     click.echo(f"also pushing to additional ECR repository: {codebase.build.additional_repository}")
```
Comment (acodeninja) [inline image_builder/commands/build.py:43]: Does this want to stay or go?

File: image_builder/publish.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    for tag in tags:
+        from_tag = f"{initial_repository}:{tag}"
+        to_tag = f"{additional_repository}:{tag}"
+        tag_cmd = ["docker", "tag", from_tag, to_tag]
+        print(f"Running command: {' '.join(tag_cmd)}")
+        run_and_check_result(tag_cmd)
+        push_cmd = ["docker", "push", to_tag]
+        print(f"Running command: {' '.join(push_cmd)}")
+        run_and_check_result(push_cmd)
```
Comment (acodeninja) [inline image_builder/publish.py:29]: I'm wondering if this tag and push is needed? Does just doing the docker login to both at the start then adding the extra repo to the list of published repos for the pack command not do the same thing but including any additional stuff pack does?

**Review by acodeninja — COMMENTED** [review]
Few thoughts and a gotcha.

File: image_builder/pack.py [hunk 3]
```diff
@@ -38,18 +39,25 @@ def build(
         if proc.returncode != 0:
             raise PackCommandFailedError
 
+        if self.codebase.build.additional_repository:
```
[the remark this answers is not in the document]
Comment (antroy-madetech) [reply image_builder/pack.py:42]: Good point.

File: image_builder/publish.py (hunk 2 above)
Comment (acodeninja) [reply image_builder/publish.py:29]: As discussed, docker multiple login is possible but pack pushing to multiple registries is not.

File: image_builder/commands/deploy.py [hunk 4]
```diff
@@ -6,6 +6,7 @@
 
 import click
 
+from image_builder.const import ECR_REPO
```
Comment (WillGibson) [inline image_builder/commands/deploy.py:9]: Manifesting my dislike of abbreviating everything...

Now we have a mixture of "repo" and repository", why not use the full name?

File: test/image_builder/commands/test_build.py [hunk 5]
```diff
@@ -53,6 +62,40 @@ def test_perfect_build(self, pack, docker, codebase, notify, progress):
...
+        self.assertIn(
+            "Using buildpacks: ['fagiani/apt', 'paketo-buildpacks/git', "
+            "'paketo-buildpacks/python', 'fagiani/run']",
+            result.output,
+        )
+
+        pack().build.assert_called()
+        pack().codebase.setup.assert_called()
+        progress().set_current_phase.assert_has_calls([])
+        notify().post_build_progress.assert_has_calls([call(ANY, ANY)] * 2)
```
Comment (WillGibson) [inline test/image_builder/commands/test_build.py:78]: ❤️  for the more rigorous asserting

**Pull request merged**